include/crypto/modes.h

```diff
@@ -56,12 +56,15 @@ typedef unsigned char u8;
                         asm ("bswapl %0"                \
                         : "+r"(ret_));   ret_;          })
 #  elif defined(__aarch64__)
-#   define BSWAP8(x) ({ u64 ret_;                       \
+#   if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
+       __BYTE_ORDER__==__ORDER_LITTLE_ENDIAN__
```

Comment:
What happens if the byte order is big endian?

Reply:
if is_endian.little is false, BSWAP8 and BSWAP4 would be not using.

Reply:
D'oh...  of course